var/spack/repos/builtin/packages/ioapi/package.py

```diff
@@ -0,0 +1,48 @@
+# Copyright 2013-2019 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import glob
+import os
+
+from spack import *
+
+
+class Ioapi(MakefilePackage):
+    """Models-3/EDSS Input/Output Applications Programming Interface."""
+
+    homepage = "https://www.cmascenter.org/ioapi/"
+    url      = "https://www.cmascenter.org/ioapi/download/ioapi-3.2.tar.gz"
+    version('3.2', sha256=('56771ff0053d47f2445e00ff369bca7b'
+                           'fc484325a2816b2c648744e523134fe9'))
```

Comment:
You can put all of this on a single line. Spack exempts checksums from its line-length checks.

Reply:
Fixed in 31ff038

Out of curiosity, how does the spack CI, etc apply the exemption?  I thought it was from the `.flake8_packages` config file in the project root, but after making your suggested change, `flake8` now raises E501:

```console
$ flake8 --config=.flake8_packages var/spack/repos/builtin/packages/ioapi/package.py
var/spack/repos/builtin/packages/ioapi/package.py:9:1: F403 'from spack import *' used; unable to detect undefined names
var/spack/repos/builtin/packages/ioapi/package.py:17:80: E501 line too long (93 > 79 characters)
```

I have the latest version of that config file:

```console
$ GIT_PAGER= git log --oneline -n 1 .flake8_packages
a750237 flake8: add exceptions for overly pedantic camelcase rules from pep8-naming (#11477)
```

```console
$ flake8 --version
3.7.8 (mccabe: 0.6.1, pycodestyle: 2.5.0, pyflakes: 2.1.0) CPython 3.7.2 on Linux
```

Reply:
Ah, nevermind.  I read the Travis log and found it runs `./share/spack/qa/run-flake8-tests` which in turn runs `spack flake8`.  So I can simply run `spack flake8 path/to/my/package.py`

Reply:
You can simply run `spack flake8` and Spack will find all files that have been changed in the current branch. Exemptions are listed [here](https://github.com/spack/spack/blob/develop/lib/spack/spack/cmd/flake8.py#L40,L84).